src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionActionServiceImpl.java

```diff
@@ -316,6 +320,33 @@ public void transferItemHold(UUID transactionId, String itemBarcode) {
     eventPublisher.publishEvent(MoveRequestEvent.of(transaction, item));
   }
 
+  @Override
+  public void finalcheckinItemHold(UUID transactionId, UUID servicePointId) {
+    var transaction = fetchTransactionById(transactionId);
+
+    verifyState(transaction, ITEM_RECEIVED, RECEIVE_UNANNOUNCED);
+
+    var loan = loanService.getById(transaction.getHold().getFolioLoanId());
+
+    if (loan.getAction().equalsIgnoreCase("checkedin") && loan.getStatus().getName().equals("closed")) {
```

Comment:
this check is already implemented. Just create a method for it and re-use here 
https://github.com/folio-org/mod-inn-reach/blob/74ca25f6cdf58500279e89f2e672ac062a033b2c/src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionActionServiceImpl.java#L201

Reply:
refactored